packages/common/directives/src/filter/filter.directive.ts

```diff
@@ -1,37 +1,54 @@
-import { Directive, ElementRef, HostListener, Input, OnDestroy, OnInit } from '@angular/core';
-import { ControlValueInterceptor, ControlValueInterceptorDescriptor } from '@angular-ru/common/forms';
+import { Directive, ElementRef, HostListener, Input } from '@angular/core';
+import { ControlValueInterceptor } from '@angular-ru/common/forms';
 import { filter, FilterPredicateFn } from '@angular-ru/common/string';
 
 @Directive({
     selector: '[filter]',
     providers: [ControlValueInterceptor]
 })
-export class FilterDirective implements OnInit, OnDestroy {
+export class FilterDirective {
     @Input() public filter: string[] | FilterPredicateFn | RegExp = [];
-    public preparedValue: string = '';
 
-    private controlValueOperator: ControlValueInterceptorDescriptor = {
-        toModelValue: (value: string): string => {
-            this.preparedValue = filter(value, this.filter);
-            return this.preparedValue;
+    private throttle: boolean = false;
+    private newValue: string = '';
+    private newEvent: InputEvent = null!;
+
+    constructor(private readonly elementRef: ElementRef<HTMLInputElement>) {}
+
+    @HostListener('input', ['$event'])
+    public onInput(event: InputEvent): boolean {
+        this.stopPropagation(event);
+
+        if (this.throttle) {
+            this.throttle = false;
+        } else {
+            this.prepareNewValue();
+            this.prepareNewEvent(event);
+            this.throttle = true;
+            this.emitNewEvent();
         }
-    };
+        return false;
+    }
 
-    constructor(
-        private readonly interceptor: ControlValueInterceptor,
-        private readonly elementRef: ElementRef<HTMLInputElement>
-    ) {}
+    private stopPropagation(event: InputEvent): void {
+        if (event instanceof InputEvent) {
+            event.preventDefault();
+            event.stopPropagation();
+            event.stopImmediatePropagation();
+        }
+    }
 
-    public ngOnInit(): void {
-        this.interceptor.attach(this.controlValueOperator);
+    private prepareNewValue(): void {
```

Comment:
make function return `string`, not assigning to `this`

Reply:
ok